app/main.py

```diff
@@ -1,25 +1,90 @@
-from abc import ABC
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.private_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> int:
+        return getattr(instance, self.private_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(
+                f"Value {value} is not within the allowed range "
+                f"[{self.min_amount}, {self.max_amount}]"
+            )
+        isinstance(instance, self.private_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
+
+    @abstractmethod
+    def is_valid(self, visitor: Visitor) -> None:
```

Comment:
There is no need for `is_valid` methods in all classes, you should define `age`, `weight` and `height` as class attributes and assign corresponding instances of `IntegerRange` class with limitations to each range. It will allow to utilize the `__set__` method later to trigger an error for further processing.